src/Lower.cpp

```diff
@@ -156,10 +156,17 @@ Module lower(const vector<Function> &output_funcs,
     debug(1) << "Computing bounds of each function's value\n";
     FuncValueBounds func_bounds = compute_function_value_bounds(order, env);
 
+    bool will_inject_host_copies =
+        (t.has_gpu_feature() ||
```

Comment:
A helper function `Target::has_device_interface` would be nice, at least to wrap up the weird Hexagon logic.